File: backend/app/schemas/publication.py
from __future__ import annotations

import uuid

from app.schemas.common import CamelModel


class PublicationRead(CamelModel):
    id: uuid.UUID
    pubmed_id: int | None = None
    doi: str | None = None
    pmc_id: str | None = None
    title: str
    abstract: str | None = None
    authors: list[str] = []
    journal: str | None = None
    year: int | None = None
    volume: str | None = None
    pages: str | None = None
    url: str | None = None


class PublicationDetail(PublicationRead):
    sequence_accessions: list[str] = []

File: frontend/src/types/publication.ts
/**
 * Scientific publication records served by `/publications`.
 * Every field mirrors the real bibliographic record (PubMed / source
 * REFERENCE blocks) — nothing is synthesised client-side.
 */
export interface Publication {
  id: string;
  pubmedId: number | null;
  doi: string | null;
  pmcId: string | null;
  title: string;
  abstract: string | null;
  authors: string[];
  journal: string | null;
  year: number | null;
  volume: string | null;
  pages: string | null;
  url: string | null;
}

export interface PublicationDetail extends Publication {
  /** Accessions of sequence records linked to this publication. */
  sequenceAccessions: string[];
}

export interface PublicationListResponse {
  results: Publication[];
  total: number;
  nextCursor: string | null;
}
